app/assets/src/components/views/discovery/DiscoveryView.jsx

```diff
@@ -130,11 +136,15 @@ class DiscoveryView extends React.Component {
         selectedSampleIds: new Set(),
         showFilters: true,
         showStats: true,
+        userDataCounts: null,
       },
       localState,
       sessionState,
       urlState
     );
+
+    this.loadUserDataStats();
```

Comment:
curious why we are not waiting on the result here... for performance in the common case? could this lead to any race condition where something shows before the empty state? maybe worth explaining in a comment

Reply:
We do not wait for any data loading. When we use await it is inside an async function. No rendering should assume that the data is loaded.